nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-pubsub-processors/src/test/java/org/apache/nifi/processors/kafka/pubsub/AbstractKafkaProcessorLifecycelTest.java

```diff
@@ -0,0 +1,253 @@
+            assertTrue(e.getMessage().contains("'topic' is invalid because topic is required"));
+        }
+
+        runner.setProperty(ConsumeKafka.TOPIC, "");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+
+        runner.setProperty(ConsumeKafka.TOPIC, "  ");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+        runner.setProperty(ConsumeKafka.TOPIC, "blah");
+
+        runner.removeProperty(ConsumeKafka.CLIENT_ID);
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("invalid because client.id is required"));
+        }
+
+        runner.setProperty(ConsumeKafka.CLIENT_ID, "");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+
+        runner.setProperty(ConsumeKafka.CLIENT_ID, "   ");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+        runner.setProperty(ConsumeKafka.CLIENT_ID, "ghj");
+
+        runner.setProperty(PublishKafka.KERBEROS_PRINCIPLE, "");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+        runner.setProperty(PublishKafka.KERBEROS_PRINCIPLE, "  ");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+    }
+
+    /*
+     * The goal of this test is to validate the correctness of
+     * AbstractKafkaProcessor's implementation of onTrigger() in a highly
+     * concurrent environment. That is:
+     * - upon processing failures (e.g., unhandled exceptions), the target Kafka
+     *   resource is reset (closed and re-created)
+     * - no FlowFile is unaccounted for. FlowFiles left in the queue and FlowFiles
+     *   in Success relationship = testCount
+     * - failed executions that did not result in the call to close/reset summed with
+     *   verified calls to close should equal total request failed
+     */
+    @Test
+    public void validateLifecycleCorrectnessWithProcessingFailures() throws Exception {
```

Comment:
I see this fairly often.


-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.apache.nifi.processors.kafka.pubsub.AbstractKafkaProcessorLifecycelTest
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.811 sec <<< FAILURE! - in org.apache.nifi.processors.kafka.pubsub.AbstractKafkaProcessorLifecycelTest
validateLifecycleCorrectnessWithProcessingFailures(org.apache.nifi.processors.kafka.pubsub.AbstractKafkaProcessorLifecycelTest)  Time elapsed: 0.694 sec  <<< FAILURE!
java.lang.AssertionError: expected:<10000> but was:<9999>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at org.apache.nifi.processors.kafka.pubsub.AbstractKafkaProcessorLifecycelTest.validateLifecycleCorrectnessWithProcessingFailures(AbstractKafkaProcessorLifecycelTest.java:221)